### ISSUE TITLE
Disabling CollisionShape3D removes it from PhysicsDirectSpaceState3D but Enabling does not add it back (until the next physics tick?)

### ISSUE THREAD

**Issue by jamie-pate** [godotengine/godot#90542]
### Tested versions

Reproducible in 4.2.1-stable, 4.3.dev5

### System information

Godot v4.3.dev5 - Ubuntu 22.04.4 LTS 22.04 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Laptop GPU () - 12th Gen Intel(R) Core(TM) i7-12700H (14 Threads)

### Issue description

Disabling CollisionShape3D removes it from PhysicsDirectSpaceState3D but Enabling does not add it back (until the next physics tick?)

This used to work in godot3.5 (Bullet physics)

Collisions before disable/enable: [{ "rid": RID(4200478015488), "collider_id": 28387050725, "collider": Area3D:<Area3D#28387050725>, "shape": 0 }]
Collisions after disable: []
Collisions after disable/enable: []
Collisions After reset state: []
Collisions after 0.1s: [{ "rid": RID(4200478015488), "collider_id": 28387050725, "collider": Area3D:<Area3D#28387050725>, "shape": 0 }]

If there is a way to manually force a physics tick that would be an acceptable workaround.

Found while attempting to port the bone rigging feature in [MagicaVoxel-Importer](https://github.com/jamie-pate/MagicaVoxel-Importer/tree/0c22f88025a8673245014291bd790b9e56ff2119) [demo project](https://github.com/jamie-pate/govox)

Related issue: https://github.com/godotengine/godot/issues/67168

### Steps to reproduce

See the minimal reproduction.

### Minimal reproduction project (MRP)

[shape-disable-enable.zip](https://github.com/godotengine/godot/files/14950469/shape-disable-enable.zip)

**Comment by ryevdokimov** [comment]
Your MRP didn't fully upload.

**Comment by jamie-pate** [comment]
> Your MRP didn't fully upload.

Oops, updated

**Comment by AThousandShips** [comment]
I'd say this is a limitation that's known and documented, see the [documentation](https://docs.godotengine.org/en/stable/classes/class_collisionshape2d.html#class-collisionshape2d-property-disabled) for the property and how it's supposed to be set deferred

**Comment by jamie-pate** [comment]
That documentation is for CollisionShape3D, should it be added to CollisionShape2D as well ?

Also it fails to mention _why_ you should use `set_deferred()`... It's likely to prevent issues like https://github.com/godotengine/godot/issues/67168, not my issue where I _do_ want immediate updates.

**Comment by jamie-pate** [comment]
WORKAROUND: (and probably the correct way to do this) is to add the shape's RIDs to the `exclude` property of `PhysicsRayQueryParameters3D`

Still unexpected that `disabled = true` works but `disabled = false` does not.. I suspect the RID gets removed from the 'database' when setting `disabled = true` but can't be re-added when you set `disabled = false` because a new physics tick needs to happen first.
